engine/src/eth/rpc.rs

```diff
@@ -260,17 +264,27 @@ impl EthSigningRpcApi for EthRpcSigningClient {
 		self.signer.address()
 	}
 
-	async fn send_transaction(&self, mut tx: Eip1559TransactionRequest) -> Result<TxHash> {
+	async fn sign_transaction(&self, mut tx: Eip1559TransactionRequest) -> Result<Bytes> {
 		tx.nonce = Some(self.get_next_nonce().await?);
+		let tx = TypedTransaction::Eip1559(tx);
 
-		let res = self.signer.send_transaction(tx, None).await;
-		if res.is_err() {
-			// Reset the nonce just in case (it will be re-requested during next broadcast)
-			tracing::warn!("Resetting eth broadcaster nonce due to error");
-			*self.nonce_info.lock().await = None;
-		}
+		self.signer
```

Comment:
I wanted to use `SignerMiddleware::sign_transaction`, but it is private. 
I think you are right about `fill_transaction`. On closer inspection, i think it might be the only place we set the nonce. Thought it was just for the chain id and address.

Reply:
Pretty sure we can skip `fill_transaction` because we set the chain id, nonce, and address ourselves, but it will not hurt to add it in. Might help in the future.

Reply:
I think using `fill_transaction` seems right to me.